pkg/ccl/serverccl/diagnosticsccl/diagnosticsccl_test: TestUsageQuantization failed

pkg/ccl/serverccl/diagnosticsccl/diagnosticsccl_test.TestUsageQuantization [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21347443?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21347443?buildTab=artifacts#/) on master @ [56ed2981ffdc045861af3c3f6e6566c7e652acb1](https://github.com/cockroachdb/cockroach/commits/56ed2981ffdc045861af3c3f6e6566c7e652acb1):


```
=== RUN   TestUsageQuantization
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/d521f634d36e98ce96ee6e72c6b25f02/logTestUsageQuantization2050219791
    test_log_scope.go:82: use -show-logs to present logs inline
    reporter_test.go:539: 
        	Error Trace:	pkg/ccl/serverccl/diagnosticsccl/reporter_test.go:539
        	Error:      	Not equal: 
        	            	expected: 10000
        	            	actual  : 1000
        	Test:       	TestUsageQuantization
        	Messages:   	quantization incorrect for query "SHOW application_name"
    panic.go:694: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/d521f634d36e98ce96ee6e72c6b25f02/logTestUsageQuantization2050219791
--- FAIL: TestUsageQuantization (4.32s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/obs-prs
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestUsageQuantization.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-63933

--- a/pkg/ccl/serverccl/diagnosticsccl/BUILD.bazel
+++ b/pkg/ccl/serverccl/diagnosticsccl/BUILD.bazel
@@ -23,6 +23,7 @@ go_test(
         "//pkg/sql",
         "//pkg/sql/catalog/lease",
         "//pkg/sql/isql",
+        "//pkg/sql/sqlstats",
         "//pkg/sql/sqlstats/persistedsqlstats/sqlstatstestutil",
         "//pkg/testutils",
         "//pkg/testutils/diagutils",

--- a/pkg/ccl/serverccl/diagnosticsccl/reporter_test.go
+++ b/pkg/ccl/serverccl/diagnosticsccl/reporter_test.go
@@ -26,6 +26,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/sql"
 	"github.com/cockroachdb/cockroach/pkg/sql/catalog/lease"
 	"github.com/cockroachdb/cockroach/pkg/sql/isql"
+	"github.com/cockroachdb/cockroach/pkg/sql/sqlstats"
 	"github.com/cockroachdb/cockroach/pkg/sql/sqlstats/persistedsqlstats/sqlstatstestutil"
 	"github.com/cockroachdb/cockroach/pkg/testutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/diagutils"
@@ -459,12 +460,21 @@ func TestUsageQuantization(t *testing.T) {
 	ctx := context.Background()
 
 	url := r.URL()
+	// Pin the SQL stats clock so that all executions land in the same
+	// aggregation window. In-memory stats are partitioned by aggregation
+	// timestamp, so a real clock that crosses an hour boundary mid-test
+	// would split a query's executions across multiple entries and break
+	// the per-entry count assertion below.
+	stubTime := timeutil.Now()
 	s, db, _ := serverutils.StartServer(t, base.TestServerArgs{
 		Knobs: base.TestingKnobs{
 			Server: &server.TestingKnobs{
 				DiagnosticsTestingKnobs: diagnostics.TestingKnobs{
 					OverrideReportingURL: &url,
 				},
+			},
+			SQLStatsKnobs: &sqlstats.TestingKnobs{
+				StubTimeNow: func() time.Time { return stubTime },
 			},
 		},
 	})

